verification/errors.go

```diff
@@ -0,0 +1,49 @@
+package verification
+
+// ErrorVerificationSkipped is used when signature verification was not performed due to "SignatureVerification" being set to "skip" in the trust policy
+type ErrorVerificationSkipped struct {
+	msg string
+}
+
+func (e ErrorVerificationSkipped) Error() string {
+	if e.msg != "" {
+		return e.msg
+	}
+	return "signature verification was skipped as specified in the trust policy"
+}
+
+// ErrorNoApplicableTrustPolicy is used when there is no trust policy that applies to the given artifact
+type ErrorNoApplicableTrustPolicy struct {
+	msg string
+}
+
+func (e ErrorNoApplicableTrustPolicy) Error() string {
+	if e.msg != "" {
+		return e.msg
+	}
+	return "there is no applicable trust policy for the given artifact"
+}
+
+// ErrorSignatureRetrievalFailed is used when notation is unable to retrieve the digital signature/s for the given artifact
+type ErrorSignatureRetrievalFailed struct {
```

Comment:
Should we differentiate this condition from when there are no signatures associated with an artifact (unsigned)? That condition also seems different than the one where we evaluated some signatures and failed.

Reply:
See `ErrorSignatureRetrievalFailed` for that purpose